docs/tutorials/forum-application/8.app.md

```diff
@@ -0,0 +1,422 @@
+# App.go
+
+*In this section you will find the code for entire `app.go` file.*
+
+```go
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/andynog/abci2-forum-app/model"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	cryptoproto "github.com/cometbft/cometbft/api/cometbft/crypto/v1"
+	cryptoencoding "github.com/cometbft/cometbft/crypto/encoding"
+
+	"github.com/cometbft/cometbft/version"
+	"github.com/dgraph-io/badger/v3"
+)
+
+const ApplicationVersion = 1
+
+type ForumApp struct {
+	abci.BaseApplication
+	valAddrToPubKeyMap map[string]cryptoproto.PublicKey
```

Comment:
How do we recover this map is we crash and recover?

Reply:
in `InitChain` there's an `updateValidator` that set's the value in the app state. So if the node crashes it retrieves from the app state in the DB, as long as the app is initialized fine, at least the genesis validator should be there. But we don't have logic to do validator updates in the tutorial. Could be a future enhancement.